packages/libs/react-ui/src/components/Grid/GridRoot.tsx

```diff
@@ -0,0 +1,56 @@
+import {
+  containerColumnVariants,
+  explicitColumnVariant,
+  gapVariants,
+  gridContainerClass,
+  ResponsiveInputType,
+} from './Grid.css';
+
+import classNames from 'classnames';
+import React, { FC, ReactNode } from 'react';
+
+export interface IGridRootProps {
+  children?: ReactNode;
+  columns?: ResponsiveInputType;
+  spacing?: keyof typeof gapVariants;
+}
+
+const GridRoot: FC<IGridRootProps> = ({
+  children,
+  columns,
+  spacing = 'md',
+}) => {
+  const classList = classNames(
+    gapVariants[spacing],
+    gridContainerClass,
+    columns && typeof columns === 'number' && explicitColumnVariant[columns],
+
+    columns &&
+      typeof columns !== 'number' &&
+      columns.sm &&
+      containerColumnVariants.sm[columns.sm],
+    columns &&
+      typeof columns !== 'number' &&
+      columns.md &&
+      containerColumnVariants.md[columns.md],
+    columns &&
+      typeof columns !== 'number' &&
+      columns.lg &&
+      containerColumnVariants.lg[columns.lg],
+    columns &&
+      typeof columns !== 'number' &&
+      columns.xl &&
+      containerColumnVariants.xl[columns.xl],
+    columns &&
+      typeof columns !== 'number' &&
+      columns.xxl &&
+      containerColumnVariants.xxl[columns.xxl],
```

Comment:
And this one too. Low-effort comment as I don't have a better solution for you right now but just checking :)

Reply:
make a small function?
`const isVariant = (columns, size) => columns &&
      typeof columns !== 'number' &&
      columns[size] &&
      containerColumnVariants[size][columns[size]],`